plugins/out_s3/s3.c

```diff
@@ -1055,8 +1057,13 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     if (chunk) {
         create_time = chunk->create_time;
     }
-    else {
+    else if (timep == NULL) {
         create_time = time(NULL);
+        flb_plg_error(ctx->ins, "Failed to get timestamp from the log record "
+                      "used for the S3 key, use the current time instead.");
+    }
+    else {
+        create_time = timep;
     }
```

Comment:
Why is create_time being set to timep? timep is the first_log_time. It has nothing to do with the file's create_time which is used by the timers.

Reply:
reverted.